src/marqo/core/models/update_documents.py

```diff
@@ -0,0 +1,29 @@
+from typing import Dict, Sequence, Any
+
+from pydantic import validator
+
+from marqo.api.exceptions import BadRequestError
+from marqo.base_model import ImmutableStrictBaseModel
+from marqo.tensor_search.enums import EnvVars
+from marqo.tensor_search.utils import read_env_vars_and_defaults_ints
+
+
+class UpdateDocumentsParams(ImmutableStrictBaseModel):
+    documents: Sequence[Dict[str, Any]]
+
+    @validator('documents')
+    def validate_docs(cls, documents):
+        doc_count = len(documents)
+
+        max_doc = read_env_vars_and_defaults_ints(EnvVars.MARQO_MAX_DOCUMENTS_BATCH_SIZE)
```

Comment:
This appears to be an API level condition. One option is to move this model to API, and just get a `Sequence[Dict[str, Any]]` in Core without using a model